sklearn/metrics/pairwise.py

```diff
@@ -1227,6 +1228,15 @@ def linear_kernel(X, Y=None, dense_output=True):
     return safe_sparse_dot(X, Y.T, dense_output=dense_output)
 
 
+@validate_params(
+    {
+        "X": ["array-like", "sparse matrix"],
+        "Y": ["array-like", "sparse matrix", None],
+        "degree": [Interval(Integral, 1, None, closed="left")],
+        "gamma": [Interval(Real, None, None, closed="neither"), None],
```

Comment:
@jeremiedbb do you think it makes sense to have a negative gamma?

Reply:
I also see that the kernel can be used in the Gaussian Process that somehow will pass an array of one element. We can add this option as a hidden one:

```suggestion
        "gamma": [
            Interval(Real, None, None, closed="neither"), None, Hidden(np.ndarray)
        ],
```

We might want alternatively to check why the Gaussian Process intends to pass a 1d array and if we can modify that. However, not supporting this case could potentially to a regression.

Reply:
Thanks, I will make corresponding changes, also to other kernels I'm working on.